Neos.ContentRepository.LegacyNodeMigration/Classes/Helpers/NodeDataLoader.php

```diff
@@ -15,15 +15,22 @@ public function __construct(
 
     public function getIterator(): Traversable
     {
+        $whereClause = [
+            'workspace' => $this->connection->quoteIdentifier("workspace") . ' = ' . $this->connection->quote("live"),
+            'path' => $this->connection->quoteIdentifier("path") . ' != ' . $this->connection->quote("/"),
+            'movedTo' => $this->connection->quoteIdentifier("movedto") . ' IS NULL ',
+            'removed' => $this->connection->quoteIdentifier("removed") . ' = ' . ($this->connection->getDatabasePlatform()->getName() === 'postgresql' ? 'FALSE' : '0')
+        ];
+
```

Comment:
IMO we should use parameters and specify the param types where necessary in order to keep this readable.
But I really wonder which part was the issue.
I assume that the following would be enough:

```php
        $query = $this->connection->executeQuery('
            SELECT
                *
            FROM
                neos_contentrepository_domain_model_nodedata
            WHERE
                workspace = "live"
                AND (movedto IS NULL OR removed=:removed)
                AND path != "/"
            ORDER BY
                parentpath, sortingindex, path
        ', ['removed' => false]);
```

Reply:
For some reason, the query needed quotes and the removed parameter was an 0 which is not working for Postgres as it expect a false

Reply:
Will test this suggestion. Thanks, @bwaidelich 👍

Reply:
If that doesn't work, we might have to add the param type as 3rd parameter
```php
', ['removed' => false], ['removed' => ParameterType::BOOLEAN]);
```

Reply:
Ah, OK! Did not know that :)

Reply:
<img width="954" alt="Screenshot 2023-03-02 at 10 00 43" src="https://user-images.githubusercontent.com/1014126/222385122-b0e0e719-05a0-4021-a588-8797fc2d028f.png">
Without the quotes, Postgres is not happy.

Reply:
Will install a local MariaDB and check if the query is still working with MySQL compatible DBs.

Reply:
```
            WHERE
                workspace = :liveWorkspaceName
```
and
```
', ['removed' => false, 'liveWorkspaceName' => 'live'], ...
```
should work in this case

Reply:
@markusguenther did you manage to make it run? I'm happy to help otherwise